cmd/oci-lifecycle-demo/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/opencontainers/runtime-tools/lifecycle"
+)
+
+func main() {
+	runtime := "runc"
+	bundle := "bundle"
+	id := "25"
+	lc, _ := lifecycle.NewLifecycle(runtime, bundle, id)
+
+	fmt.Println("create")
+	out, err := lc.Operate(lifecycle.LifecycleCreate)
+	fmt.Println(string(out), err)
+
+	fmt.Println("state")
+	out, err = lc.Operate(lifecycle.LifecycleState)
+	fmt.Println(string(out), err)
+
+	fmt.Println("start")
+	out, err = lc.Operate(lifecycle.LifecycleStart)
+	fmt.Println(string(out), err)
+
+	fmt.Println("state")
+	out, _ = lc.Operate(lifecycle.LifecycleState)
+	fmt.Println(string(out))
+
+	fmt.Println("delete")
+	out, _ = lc.Operate(lifecycle.LifecycleDelete)
+	fmt.Println(string(out))
+
+	fmt.Println("state")
+	out, _ = lc.Operate(lifecycle.LifecycleState)
+	fmt.Println(string(out))
```

Comment:
The sharness analog to this file is:

```shell
test_expect_success 'Demo lifecycle' "
  subreaper-wrapper ${RUNTIME} create --bundle bundle 25 &
  while test ! -e 25-created; do sleep 1; done &&  # this would be easier with runtime-spec#508, but works as a stop-gap with subreaper-wrapper touching this file when 'create' exits
  ${RUNTIME} state 25 &&
  ${RUNTIME} start 25 &&
  ${RUNTIME} state 25 &&
  ${RUNTIME} delete 25 &&
  ${RUNTIME} state 25 &&  # this line will probably fail.  I'm not sure if/how you handle that in your Go demo
  wait
"
```

I think that's more compact/obvious/flexible, although I haven't implemented `subreaper-wrapper` (although this PR does not setup subreaper handling either).